File: common/src/test/java/cc/turtl/turtlshell/junit/BootstrapMinecraftExtension.java
package cc.turtl.turtlshell.junit;

import net.minecraft.SharedConstants;
import net.minecraft.server.Bootstrap;
import org.jspecify.annotations.NonNull;
import org.junit.jupiter.api.extension.BeforeAllCallback;
import org.junit.jupiter.api.extension.ExtensionContext;

/**
 * JUnit extension that bootstraps vanilla Minecraft before any tests in the class run.
 * <p>
 * Use via {@link SetupTestDependencies} - do not reference this class directly.
 */
public class BootstrapMinecraftExtension implements BeforeAllCallback {

    @Override
    public void beforeAll(@NonNull ExtensionContext context) {
        SharedConstants.tryDetectVersion();
        Bootstrap.bootStrap();
    }
}

File: common/src/test/kotlin/cc/turtl/turtlshell/junit/BootstrapMinecraftExtension.kt
package cc.turtl.turtlshell.junit

import net.minecraft.SharedConstants
import net.minecraft.server.Bootstrap
import org.junit.jupiter.api.extension.BeforeAllCallback
import org.junit.jupiter.api.extension.ExtensionContext

/**
 * JUnit extension that bootstraps vanilla Minecraft before any tests in the class run.
 *
 *
 * Use via [SetupTestDependencies] - do not reference this class directly.
 */
class BootstrapMinecraftExtension : BeforeAllCallback {
    override fun beforeAll(context: ExtensionContext) {
        SharedConstants.tryDetectVersion()
        Bootstrap.bootStrap()
    }
}